subprojects/file-watching/src/main/java/org/gradle/internal/watch/vfs/impl/WatchingVirtualFileSystem.java

```diff
@@ -55,6 +56,7 @@
     private final DelegatingDiffCapturingUpdateFunctionDecorator delegatingUpdateFunctionDecorator;
     private final AtomicReference<FileHierarchySet> producedByCurrentBuild = new AtomicReference<>(DefaultFileHierarchySet.of());
     private final Predicate<String> watchFilter;
+    private final Set<File> buildRootDirectories = new HashSet<>();
```

Comment:
Should this be `watchedBuildRootDirectories`?

Reply:
The thing is that the directories are not watched necessarily. And the watching virtual file system needs to collect them even when file watching is disabled.

`buildRootDirectoriesForWatching` maybe?

Reply:
Sounds good.

Reply:
Renamed.